ingestion_tools/scripts/common/mesh_converter.py

```diff
@@ -15,21 +17,91 @@ def convert_mesh_to_glb(convert_func: typing.Callable[[str], trimesh.Trimesh]) -
         mesh.export(output_file)
 
     return wrapper
 
 
+def check_mesh_name(input_file: str, name: str) -> str:
+    with h5py.File(input_file, "r") as fp:
+        for mesh_list in fp["segment_list/"].keys(): # noqa
+            # Mesh name
+            mesh_name_raw: bytes = fp[f"segment_list/{mesh_list}/biological_annotation/name"][()]
+            if not mesh_name_raw:
+                logger.warning("No mesh name found at segment_list/{mesh_list} in the hff file")
+            else:
+                mesh_name: str = mesh_name_raw.decode()
+                if mesh_name == name:
+                    return mesh_list
+        raise ValueError(f"Mesh name {name} not found in the hff file")
+
+
+@convert_mesh_to_glb
+def from_hff(input_file: str, name: str) -> trimesh.Trimesh:
+    """
+    Convert an hff file to one or more glb file. More information about this format can be found
+    here: https://emdb-empiar.github.io/EMDB-SFF/
+
+    Each group mesh in the hff file is stored as a separate glb file. The name of the mesh file is the biological
+    annotation name suffixed with '.glb'.
+
+    :param input_file: the path to an hff file.
+    :param name: the name of the mesh to extract. The name is the biological annotation name.
+    :return: A list of meshes with the file extension .glb
+    """
+
+    def _extract(_f, path) -> np.ndarray:
+        data_path = f"{path}/data"
+        # The data is stored as a string of base64 encoded bytes, so we need to decode them
+        data = base64.b64decode(fp[data_path][()])
+
+        mode_path = f"{path}/mode"
+        mode = fp[mode_path][()].decode("utf-8")
+        dtype = getattr(np, mode)
+
+        return np.frombuffer(data, dtype=dtype).reshape(-1, 3)
+
+    mesh_index = check_mesh_name(input_file, name)
+    # Open the HDF5 file
+    with h5py.File(input_file, "r") as fp:
+        # All the meshes are stored in the 'segment_list'Explore & Visualize HDF5 Files
+        logger.info(f"Schema version: {fp['version'][()]}")  # noqa
+        # Mesh name
+        logger.info(f"Processing mesh: {name}")  # noqa
+        # Color is a 4-tuple of RGBA values, 0-1 range (normalized from 0-255)
+        color = fp[f"segment_list/{mesh_index}/colour"][()]
+        meshes = []
+        for mesh in fp[f"segment_list/{mesh_index}/mesh_list/"].keys(): # noqa
+            # Create a trimesh object from the vertices and triangles data
+            triangle_path = f"segment_list/{mesh_index}/mesh_list/{mesh}/triangles"
+            triangles = _extract(fp, triangle_path)
+
+            # Do the same for the vertices
+            vertices_data_path = f"segment_list/{mesh_index}/mesh_list/{mesh}/vertices"
+            vertices = _extract(fp, vertices_data_path)
+
+            # align the vertices and triangles
+
+
+            # Create a trimesh object from the vertices and triangles data
+            mesh = trimesh.Trimesh(vertices=vertices, faces=triangles, face_colors=color)
```

Comment:
As we don't add color to the other trimeshes we create, this might be inconsistent. Also, we might not retain the same color in the neuroglancer visualization, that could create additional confusion.

Reply:
removed